common/faceting.js

```diff
@@ -608,31 +735,23 @@
                                 if (scope.facetColumn.sourceReference.uri !== uri) {
                                     defer.resolve(false);
                                 } else {
-
-                                    // console.log("Facet " + scope.facetColumn.displayname.value + " min/max: ", response);
-                                    if (scope.facetColumn.column.type.rootName.indexOf("timestamp") > -1) {
-                                        // convert and set the values if they are defined.
-                                        // if values are null, undefined, false, 0, or '' we don't want to show anything
-                                        if (response[0] && response[1]) {
-                                            var minTs = moment(response[0]);
-                                            var maxTs = moment(response[1]);
-
-                                            scope.rangeOptions.absMin = {
-                                                date: minTs.format('YYYY-MM-DD'),
-                                                time: minTs.format('hh:mm:ss')
-                                            };
-                                            scope.rangeOptions.absMax = {
-                                                date: maxTs.format('YYYY-MM-DD'),
-                                                time: maxTs.format('hh:mm:ss')
-                                            };
-                                        }
-                                    } else {
-                                        scope.rangeOptions.absMin = response[0];
-                                        scope.rangeOptions.absMax = response[1];
+                                    // initiailize the min/max values
+                                    setRangeMinMax(response[0], response[1]);
+
+                                    // if - the max/min are null
+                                    //    - bar_plot in annotation is 'false'
+                                    //    - histogram not supported for column type
+                                    if (!scope.showHistogram()) {
+                                        return;
```

Comment:
Have you tried this? I think you have to return a promise. Instead of `return;` you can just `defer.resolve(true)` here which will resolve the outer promise and end the execution.